visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -52,18 +52,38 @@ export class FileChooserController {
 					}
 
 					if (readFiles === element.files.length) {
-						this.setNewData()
+						this.setNewData(element.files)
 					}
 				}
 			}
 		})
 	}
 
-	setNewData() {
+	setNewData(sortingArray?) {
+		if (sortingArray !== undefined) {
+			this.sortFiles(sortingArray, this.files)
+		}
 		this.codeChartaService.loadFiles(this.files)
 		this.files = []
 	}
 
+	private findInFileList = (fileName: string, fileList: FileList): number => {
+		for (let index = 0; index < fileList.length; index++) {
+			if (fileList.item(index).name === fileName) {
+				return index
+			}
+		}
+		return 0
+	}
+
+	// this will sort the array based on the order the files have been imported, needed for e2e fileChooser test
+	private sortFiles(sortingArray, arrayToSort) {
+		const sorter = (a, b) => {
+			return this.findInFileList(a.fileName, sortingArray) - this.findInFileList(b.fileName, sortingArray)
+		}
+		return arrayToSort.sort(sorter)
+	}
```

Comment:
We do not have to sort anything after reading the data. We already know the insertion order and that's what we should work with.

We change the [`for of` loop to a regular loop](https://github.com/MaibornWolff/codecharta/pull/1694/files#diff-6d89f7e88a8827fe5c40c29134c21a0310e96fe2f03b575350a1f1eb3942e566R23):

```js
for (let i = 0; i < element.files.length; i++) {
  const file = elements.files[i]
```

That index is then passed to [`addNameDataPair`](https://github.com/MaibornWolff/codecharta/pull/1694/files#diff-6d89f7e88a8827fe5c40c29134c21a0310e96fe2f03b575350a1f1eb3942e566R51):

```js
this.addNameDataPair(file, content, i)
```

That is then used to [add the file at the correct index](https://github.com/MaibornWolff/codecharta/pull/1694/files#diff-6d89f7e88a8827fe5c40c29134c21a0310e96fe2f03b575350a1f1eb3942e566R100):

```js
this.files[i] = { ... }
```

Reply:
n1 :)